### PR TITLE
testing api

### PR DESCRIPTION
**Pull request by tbenr**
help replicating mainnet issue

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    request.header(CACHE_CONTROL, CACHE_NONE);
+
+    Checkpoint checkpoint =
+        new Checkpoint(
+            request.getPathParameter(EPOCH_PARAMETER),
+            request.getPathParameter(BLOCK_ROOT_PARAMETER));
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java]: if we're logging this anyway we could make it final and re-use in the `respondAsync`...

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    request.respondAsync(
+        chainDataProvider
+            .retrieveCheckpointState(
+                new Checkpoint(
+                    request.getPathParameter(EPOCH_PARAMETER),
+                    request.getPathParameter(BLOCK_ROOT_PARAMETER)))
+            .thenApply(
+                state -> {
+                  LOG.info("response: state at slot: " + state.map(BeaconState::getSlot));
+                  return AsyncApiResponse.respondWithCode(SC_OK);
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java]: is it worth responding with the time taken for the request, or logging it? just a thought

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java (hunk 1 above)
Comment (tbenr) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java]: I did it with no intention to merge it

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java (hunk 2 above)
Comment (tbenr) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java]: ye, could be useful

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java (hunk 1 above)
Comment (tbenr) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/beacon/GetCheckpointState.java]: ah yes.. copy paste error :)

**Comment by tbenr** [comment]
closing this since it was needed to replicate some state regeneration issues.

**Pull request closed without merging**